Google.PowerShell/CloudResourceManager/GoogleCloudProjectCmdlets.cs

```diff
@@ -0,0 +1,133 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Google.PowerShell.CloudResourceManager
+{
+    /// <summary>
+    /// <para type="synopsis">
+    /// Retrieves one or more Google Cloud projects.
+    /// </para>
+    /// <para type="description">
+    /// Retrieves one or more Google Cloud projects. The cmdlet will return all available projects if no parameter
+    /// is used. If -Name, -Id or -Label is used, the cmdlets will return the projects that match the given arguments.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> Get-GcProject</code>
+    ///   <para>This command gets all the available Google Cloud projects.</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GcProject -Name "My Project"</code>
+    ///   <para>This command gets the project that has the name "My Project".</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GcProject -Id "my-id"</code>
+    ///   <para>This command gets the project that has the Id "My Project".</para>
+    /// </example>
+    /// <example>
+    ///   <code>PS C:\> Get-GcProject -Label {"environment" = "test"}</code>
+    ///   <para>This command gets all the projects that has the label "environment" with value "test".</para>
+    /// </example>
+    /// <para
+    ///     type="link"
+    ///     uri="(https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy#cloud_platform_projects)">
+    /// [Projects]
+    /// </para>
+    /// <para type="link" uri="(https://cloud.google.com/resource-manager/docs/using-labels)">
+    /// [Labels]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "GcProject")]
+    public class GetGcProjectCmdlet : CloudResourceManagerCmdlet
+    {
+        /// <summary>
+        /// <para type="description">
+        /// The name of the project to seach for.
+        /// This parameter is case insensitive.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Id of the project to seach for.
+        /// This parameter is case insensitive.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public string ProjectId { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The labels of the project to seach for.
+        /// Key and value of the label should be in lower case with no spaces in them.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ValidateNotNullOrEmpty]
+        public Hashtable Label { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            // Filter is case insensitive.
+            string filter = string.Empty;
```

Comment:
Split this into a couple of functions? One to get the filter, one make the request and return an enum of `Project`s? End with a call to `WriteObject(projectsEnum, true);`?